### PR TITLE
Clean up the bazel jobs.

### PR DESCRIPTION
**Pull request by spxtr**
Explicitly set `--service-account` and run the `JOB_NAME` job. This way we don't need the `git clone ... && bootstrap` in the yaml file.

### CODE REVIEW

File: prow/presubmit.yaml [hunk 1]
```diff
@@ -180,9 +178,7 @@ kubernetes/test-infra:
   trigger: "@k8s-bot (bazel )?test this"
   spec:
     containers:
-    - image: gcr.io/k8s-testimages/bazelbuild:0.2
-      command: ["/bin/bash", "-c"]
-      args: ["git clone https://github.com/kubernetes/test-infra && ./test-infra/jenkins/bootstrap.py --upload='gs://kubernetes-jenkins/pr-logs' --repo=k8s.io/test-infra --pull=${PULL_REFS} --job=pull-test-infra-bazel"]
```
Comment (fejta) [inline prow/presubmit.yaml]: Yay!

File: images/pull-kubernetes-bazel/runner [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+git clone https://github.com/kubernetes/test-infra
```
Comment (fejta) [inline images/pull-kubernetes-bazel/runner]: You could put the "check out the PR w/ retry" here.

File: images/pull-kubernetes-bazel/runner [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+./test-infra/jenkins/bootstrap.py \
+    --upload='gs://kubernetes-jenkins/pr-logs' \
+    --repo=k8s.io/kubernetes \
+    --pull=${PULL_REFS} \
```
Comment (fejta) [inline images/pull-kubernetes-bazel/runner]: This feels a little odd...

I would prefer we a) make this a python program b) use argparse to specify --pull, etc and c) pass in JOB_NAME and PULL_REFS as arguments to the image.

AKA I would like to move away from magic environment variables to a more explicit interface using arguments.

File: images/pull-kubernetes-bazel/runner (hunk 2 above)
Comment (spxtr) [inline images/pull-kubernetes-bazel/runner]: Sure, in a future PR.

File: images/pull-kubernetes-bazel/runner (hunk 3 above)
Comment (spxtr) [inline images/pull-kubernetes-bazel/runner]: I think the contract between prow and the pods it spawns are that it sets the small set of environment variables that it knows. Job name, PR number, branch, and that's about it. I don't think that's too gross, although we can certainly pass in some of them as arguments.

Comment (fejta) [inline images/pull-kubernetes-bazel/runner]: Right, that makes sense, but I'm not sure if we want to keep this pattern or move away from it... The benefit is that it is portable but the downside is hard to understand.

**Pull request merged**